pkg/catalog/configmeta.go

```diff
@@ -116,6 +136,83 @@ func servicesForRole(
 	return result
 }
 
+// genconfigConnections will look at the cluster spec
+// and generates a map of configmap type and corresponding
+// configmaps to be connected to the given cluster
+func genconfigConnections(
+	cr *kdv1.KubeDirectorCluster,
+) (map[string]map[string]map[string]string, error) {
+
+	cmMap := make(map[string]map[string]string)
+	kdcm := make(map[string]map[string]map[string]string)
+	for _, connectedCmName := range cr.Spec.Connections.ConfigMaps {
+		cm, err := observer.GetConfigMap(cr.Namespace, connectedCmName)
+		if kdConfigMapType, ok := cm.Labels["kubedirectorcmtype"]; ok {
```

Comment:
We shouldn't hardcode this label key string; it should be defined as a constant in one of our types.go files.

Also, the label key should be a domain-ified thing that makes use of KdDomainBase. Check some of the other label keys for examples, like ClusterAppLabel.